lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -1096,9 +1098,11 @@ instance ToJSON ApiByronWalletBalance where
     toJSON = genericToJSON defaultRecordTypeOptions
 
 instance FromJSON (ApiT PoolId) where
-    parseJSON = parseJSON >=> eitherToParser . bimap ShowFmt ApiT . fromText
+    parseJSON = parseJSON >=> eitherToParser
+           . bimap ShowFmt ApiT
+           . decodePoolIdBech32
```

Comment:
Please provide a formatter or an editor config that works. Otherwise these issues will pop up over and over again. I'm not counting spaces while I code.

Reply:
@hasufell wrote:
> Please provide a formatter or an editor config that works. Otherwise these issues will pop up over and over again. 

**Regarding automated formatting:** Unfortunately our team doesn't use an automated formatter besides [stylish-haskell](https://github.com/jaspervdj/stylish-haskell) (for imports), but if you feel strongly that we should do, then you could start a discussion in our team channel, and try to convince the team of the benefits. :+1: 

**Regarding an editor config:** We do provide one [here](https://github.com/input-output-hk/cardano-wallet/blob/master/.editorconfig).

@hasufell wrote:
> I'm not counting spaces while I code.

Our [team coding standard](https://github.com/input-output-hk/adrestia/wiki/Coding-Standards) is actually rather simple in this respect: we just indent each level by 4 spaces. (We make an exception for `where` clauses, which are indented by 2 spaces.)

Are you facing an issue that makes it hard to indent in this way? If so, we can discuss ways to fix the issue.

Reply:
> Regarding automated formatting: Unfortunately our team doesn't use an automated formatter besides stylish-haskell (for imports), but if you feel strongly that we should do, then you could start a discussion in our team channel, and try to convince the team of the benefits. +1

I don't have a strong opinion, except that I avoid formatting discussions as much as possible. I'm just saying these things will pop up again if the tooling (formatter or config) doesn't work consistently. I don't think about indentation while coding and I am not counting spaces while doing so.

I also believe developers time is too valuable to count spaces in pull requests (your time and my time). I'll do whatever formatting you want, as long as I'm provided with the tools to do so.

> Regarding an editor config: We do provide one here.

I'm using that one and it seems it doesn't work consistently. Help appreciated. Here is my editor config https://gogs.hasufell.de/hasufell/vim-config/src/branch/workman

> Are you facing an issue that makes it hard to indent in this way? If so, we can discuss ways to fix the issue.

See above.

Reply:
@hasufell wrote:
> I'll do whatever formatting you want, as long as I'm provided with the tools to do so.

I'm sorry, I don't have an automated tool to provide you with, otherwise I would. As I mentioned before, if there is a tool that you think we could use, then feel free to discuss it in the team channel. I would happy to help with the discussion.

> I don't think about indentation while coding and I am not counting spaces while doing so.
> I also believe developers time is too valuable to count spaces in pull requests (your time and my time). 

The reason for having a team standard is so that we don't have to waste time during PR reviews debating code style. If we have to discuss this in every PR, it would certainly waste time.

As far as I know, most people in our team just configure their tab key to emit 4 spaces. Then no counting is required. Could you do that?

Thanks!

Reply:
@hasufell I had a look at your vim config and... it's .. a lot. Looks quite complicated and there's probably some plugin that you have which is overriding the editorconfig plugin. 

I could share mine if you want but I am not sure it's going to help you much. 

I do agree that arguing about white-spaces over a PR is pointless, but it should just not happened that code is wrongly indented because your editor should be configured to pick up the right config. 
So far, we've all been using that `.editorconfig` without any issues and with various editors (vim, emacs and sublime at least). I can confirm it works perfectly fine with vim 7 & 8.

Reply:
> Looks quite complicated and there's probably some plugin that you have which is overriding the editorconfig plugin.

It is not overridden, because when I press tab it indents 4 spaces. It's possible that there is diverging behavior on pressing `Enter`. I expect an editor to do the right thing when I press enter and not have to manually indent every line I type. I'm not sure what the correct configuration for that is though.

Reply:
@hasufell wrote:
> So, can someone say what the indentation policy for those various constructs is?

Which constructs?

Reply:
@hasufell Just in case there's any doubt, we generally indent code with:

- 4 spaces per level of nesting.
- 2 spaces for a `where` clause.

**Example:**
```hs
launchMissiles trace a b c = do
    missiles <- prepareMissiles
    forM_ missiles $ \missile -> do
        log trace Warning $ MsgMissileLaunch missile        
        launchMissile missile
    pure missiles    
  where
    launchMissile m =
        inializeLauncherModule m
        lightFuse m
    prepareMissiles = do
        missiles <- enumerateMissiles
        verifyMissilesIntact missiles
        pure missiles
```

Reply:
So my configuration for haskell-vim now is

```vim
let g:haskell_indent_if = 0
let g:haskell_indent_case = 4
let g:haskell_indent_let = 4
let g:haskell_indent_where = 6
let g:haskell_indent_before_where = 2
let g:haskell_indent_after_bare_where = 2
let g:haskell_indent_do = 4
let g:haskell_indent_in = 0
let g:haskell_indent_guard = 2
```

This doesn't play nicely with `do` I believe. See below.

----

* do

```hs
foo :: IO Int
foo = do
    x <- pure 1
    do
        pure (2 + x)
```

* if

```hs
foo :: IO Int
foo =
    if True
    then pure 3
    else pure 2
```

* case

```hs
foo :: IO Int
foo =
    case True of
        True -> 3
        False -> 2
```

* let

```hs
foo :: IO Int
foo =
    let x = 3 in
        x + 1

foo :: IO Int
foo =
    let x = 1
    in x + 1
```

* guard

```hs
foo :: IO Int
foo
  | otherwise = 2
```

* where

```hs
foo :: IO Int
foo = x + 1
  where
    x = 3
```

So please either confirm this configuration or suggest an alternative.

Reply:
I don't see any problem with `do` above ? Also, guards should be 4-space indented like the rest

```
let g:haskell_indent_guard = 4
```

Reply:
The problem with do is if you write something on the same line as do:

```hs
foo :: IO Int
foo = do
    x <- pure 1
    do y <- pure 2
        pure (x + y)
```

Reply:
That is arguably a horrible choice of formatting and It's perhaps much better than the compiler fails to parse this :)